pkg/api/util.go

```diff
@@ -146,7 +149,7 @@ var preMapHooks = map[string]func(v string) (string, error){
 //
 // In case of parsing error, a new parseError is returned to the caller.
 // The caller can use the Unwrap method to get the original error.
-func mapStructure(input, output interface{}) (err error) {
+func mapStructure(input, output interface{}, s *Service) (err error) {
```

Comment:
This is a generic function, it should not take a `Service` object as a param, but the object should be wrapped in a pre-hook closure.

Reply:
thankyou, yeah makes sense, on it already